tests/neg/i14983.scala

```diff
@@ -0,0 +1,13 @@
+sealed trait Tree[+A]
+final case class Leaf[+B](v: B)        extends Tree[B]
+final case class Node[+C](xs: List[C]) extends Tree[List[C]]
```

Comment:
Change this to

```suggestion
final case class Node[+C <: AnyRef](xs: List[C]) extends Tree[List[C]]
```

and use Integer instead of Int, and it's goes back to breaking